src/__tests__/search-and-replace-test.jsx

```diff
@@ -0,0 +1,185 @@
+var assert = require("assert");
+var StatefulEditorPage = require("../stateful-editor-page.jsx");
+var EditorPage = require("../editor-page.jsx");
+var ArticleEditor = require("../article-editor.jsx");
+var TestUtils = React.addons.TestUtils;
+
+var itemJson = require("./test-items/search-and-replace-item.json");
+var particleJson = require("./test-items/search-and-replace-particle.json");
+
+describe("Search and Replace", function() {
+    var editor, editorProps;
+    var searchInput, replaceInput;
+    var previousButton, nextButton, replaceButton, replaceAllButton;
+
+    beforeEach(function () {
+        var problemNum = Math.floor(Math.random() * 100);
+        var enabledFeatures = {
+            highlight: true,
+            toolTipFormats: true,
+            useMathQuill: true
+        };
+
+        editorProps = {
+            problemNum: problemNum,
+            enabledFeatures: enabledFeatures,
+            developerMode: true,
+            imageUploader: function(image, callback) {
+                _.delay(callback, 1000, "http://fake.image.url");
+            },
+            apiOptions: {
+                fancyDropdowns: true,
+                __onInputError: function() {
+                    var args = _.toArray(arguments);
+                    console.log.apply(console, ["onInputError:"].concat(args));
+                    return true;
+                },
+                __interceptInputFocus: function() {
+                    var args = _.toArray(arguments);
+                    console.log.apply(console, ["interceptInputFocus:"].concat(args));
+                    return;
+                },
+                onFocusChange: function(newPath, oldPath) {
+                    console.log("onFocusChange", newPath, oldPath);
+                },
+                __staticRender: true
+            }
+        };
+    });
+
+    describe("EditorPage", function() {
+        beforeEach(function() {
+            editorProps = $.extend(true, editorProps, itemJson);
+            editorProps.componentClass = EditorPage;
+
+            var StatefulEditorPageFactory = React.createFactory(StatefulEditorPage);
+            editor = React.render(
+                StatefulEditorPageFactory(editorProps, null),
+                document.body
+            );
```

Comment:
We should be able to use [React's new spread attributes syntax](http://facebook.github.io/react/docs/jsx-spread.html) here:

```javascript
    editor = React.render(
        <StatefulEditorPage {...editorProps} />,
        document.body
    );
```